src/rust/engine/src/nodes.rs

```diff
@@ -381,6 +417,103 @@ impl WrappedNode for Select {
   }
 }
 
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct HttpRequester(Key);
+
+impl From<HttpRequester> for NodeKey {
+  fn from(n: HttpRequester) -> Self {
+    NodeKey::HttpRequester(n)
+  }
+}
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct EngineHttpResponse {
+  response_code: Option<u16>,
+  bytes: Option<Vec<u8>>,
+  url: String,
+  headers: Vec<(String, String)>
+}
+
+impl WrappedNode for HttpRequester {
+  type Item = EngineHttpResponse;
+
+  fn run(self, context: Context) -> NodeFuture<EngineHttpResponse> {
+    use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
+
+    let value = externs::val_for(&self.0);
+    let url_string = externs::project_str(&value, "url");
+    let headers =
+      try_future!(externs::project_tuple_encoded_map(&value, "headers").map_err(|err| throw(&err)));
+
+    let mut header_map = HeaderMap::new();
+    for (k, v) in headers.iter() {
+      let header_name = HeaderName::from_bytes(k.as_bytes()).unwrap();
+      let header_value = HeaderValue::from_str(v).unwrap();
+      header_map.insert(header_name, header_value);
+    }
+
+    let url = try_future!(Url::parse(&url_string).map_err(|err| throw(&format!(
+      "HttpRequest - error parsing URL {}: {}",
+      url_string, err
+    ))));
+
+    let core = context.core.clone();
+    let request = core.http_client.get(url).headers(header_map);
+
+    let send_future = request.send();
+
+    // The Ok variant of this result is reqwest::Response, but for some reason actually importing
+    // that and trying to use it here breaks, even though it doesn't break with `reqwest::Error`.
+    send_future
+      .then(|result: Result<_, reqwest::Error>| match result {
+        Ok(resp) => {
+          let response_code = Some(resp.status().as_u16());
+          let url = resp.url().as_str().to_string();
+          let headers: Vec<(String, String)> = resp.headers()
+            .into_iter().map(|(k, v)| (k.as_str().to_string(), v.to_str().unwrap().to_string())).collect();
```

Comment:
The `unwrap` here should go away somehow (either by treating headers as bytes, by using a lossy conversion, or mostly likely by returning an error if the value was invalid)